File: ProductAPI/Models/ProductImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace ProductAPI.Models
{
    public class ProductImage
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string ProductImageId { get; set; }
        public string ProductImageName { get; set; }
        public string ProductImageUrl { get; set; }
        [JsonIgnore]
        public Product Product { get; set; }
        [ForeignKey("Product")]
        public string ProductId { get; set; }
    }
}

File: jeanstationfrontend/src/app/models/ProductImage.ts
import Product from "./Product";

export default class ProductImage{
    productImageId?: string;
    productImageName: string;
    productImageUrl: string;
    product: Array<Product>;
    productId: string;

    constructor(){
        this.productImageId = "";
        this.productImageName = "";
        this.productImageUrl = "";
        this.product = [];
        this.productId = "";
    }

}